File: API-Samples/Java/app/src/main/java/videoindexersamples/authentication/AccessTokenRequest.java
package videoindexersamples.authentication;


/**
 * The request model for getting Video Indexer Access Token
 */
public class AccessTokenRequest {
    public String videoId;
    public String projectId;
    public ArmAccessTokenPermission permissionType;
    public ArmAccessTokenScope scope;

    public AccessTokenRequest(String videoId, String projectId, ArmAccessTokenPermission permission, ArmAccessTokenScope scope) {
        this.videoId = videoId;
        this.projectId = projectId;
        this.permissionType = permission;
        this.scope = scope;
    }


}

File: BringYourOwn-Samples/Src/VideoIndexerClient/auth/AccessTokenRequest.cs
using System.Text.Json.Serialization;
#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.

namespace VideoIndexerClient.auth
{
    public class AccessTokenRequest
    {
        [JsonPropertyName("permissionType")]
        public ArmAccessTokenPermission PermissionType { get; set; }

        [JsonPropertyName("scope")]
        public ArmAccessTokenScope Scope { get; set; }

        [JsonPropertyName("projectId")]
        public string ProjectId { get; set; }

        [JsonPropertyName("videoId")]
        public string VideoId { get; set; }
    }

    public class GenerateAccessTokenResponse
    {
        [JsonPropertyName("accessToken")]
        public string AccessToken { get; set; }
    }
}
